src/select/select.tsx

```diff
@@ -34,6 +39,14 @@ class Select<T> extends React.Component<Props<T>, State> {
         this.handleOptionClick = this.handleOptionClick.bind(this);
         this.toggleOptionsList = this.toggleOptionsList.bind(this);
         this.collapseOptionsList = this.collapseOptionsList.bind(this);
+        this.filterOptions = this.filterOptions.bind(this);
+        this.searchInputRef = React.createRef();
+    }
+
+    public componentDidMount() {
+        this.setState({
+            options: this.props.options
```

Comment:
Can't you just do that in the constructor instead of assigning empty array first?

Reply:
I agree.